### DISCUSSION TITLE
How to use parameters with CSF3 and Typescript correctly?

### DISCUSSION THREAD

**Discussion by mcmxcdev** [storybookjs/addon-designs#197] [Q&A] [answered]
We are using CSF3 and wondering how to satifsy Typescript in this case:

```
...

export default meta;
type Story = StoryObj<typeof meta>;

export const Default = {
  args: {
    children: 'Next',
    disabled: false,
    uppercase: false,
  },
} satisfies Story;

Default.parameters = {
  design: {
    type: 'figma',
    url: '...',
  },
};
```

This will produce an error like `Property 'parameters' does not exist on type '{ args: { children: string; disabled: boolean; uppercase: boolean; }; }'.`

It's possible to do `(Default as Story).parameters` but that's quite verbose imo.

We can also do `meta.parameters`, but then we wouldn't be able to have a different figma file linked per component variation.

**Comment by pocka** [comment, marked as answer]
https://storybook.js.org/docs/react/writing-stories/parameters

```ts
export const Default = {
  args: {
    children: 'Next',
    disabled: false,
    uppercase: false,
  },
  parameters: {
    design: {
      type: 'figma',
      url: '...',
    },
  },
} satisfies Story;
```
